commit 1e9a87038f5bc0a65a32da476e324864c23fcc8d
Author: John Levon <john.levon@joyent.com>
Date:   2019-04-02T08:50:30+00:00 (6 months ago)
    
    OS-7566 Clean up kvm-cmd usage of ctf tools
    OS-7639 ctfconvert -i option is mis-handled
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
    Approved by: Jerry Jelinek <jerry.jelinek@joyent.com>

diff --git a/Makefile.target b/Makefile.target
index 2050146..b5d2a5a 100644
--- a/Makefile.target
+++ b/Makefile.target
@@ -1,5 +1,5 @@
 # Copyright 2011 QEMU contributors
-# Portions Copyright 2011 Joyent, Inc.
+# Portions Copyright 2019 Joyent, Inc.
 # -*- Mode: makefile -*-
 
 GENERATED_HEADERS = config-target.h
@@ -382,9 +382,8 @@ ifeq ($(TRACE_BACKEND),dtrace)
 ifneq ($(strip $(CONFIG_SOLARIS)),)
 $(QEMU_PROG): $(obj-y) $(obj-$(TARGET_BASE_ARCH)-y)
 	$(call quiet-command, dtrace $(CONFIG_DTRACE_FLAGS) -o ../trace-dtrace.o -s ../trace-dtrace.dtrace -G $(obj-y) $(obj-$(TARGET_BASE_ARCH)-y),"  LINK  $(TARGET_DIR)$@.dtrace")
-	$(call CTFCONVERT_CMD)
 	$(call LINK,$(obj-y) $(obj-$(TARGET_BASE_ARCH)-y) ../trace-dtrace.o)
-	$(call CTFMERGE_CMD, $(obj-y) $(obj-$(TARGET_BASE_ARCH)-y))
+	$(call CTFCONVERT_CMD, $(obj-y) $(obj-$(TARGET_BASE_ARCH)-y))
 else
 $(QEMU_PROG): $(obj-y) $(obj-$(TARGET_BASE_ARCH)-y)
 	$(call LINK,$(obj-y) $(obj-$(TARGET_BASE_ARCH)-y))
diff --git a/ctf.sh b/ctf.sh
deleted file mode 100755
index 11fc139..0000000
--- a/ctf.sh
+++ /dev/null
@@ -1,29 +0,0 @@
-#!/usr/bin/bash
-#
-# We need to run ctfconvert on all the .o files in qemu. However, some of these
-# .o files contain some snippets that are going to cause ctfconvert to fail. If
-# ctfconvert is run with the -i option, it will delete the .o file. This is bad.
-# Instead we end up using a temporary file and move over it. 
-#
-
-sh_arg0=$(basename $0)
-
-function fail
-{
-        local msg="$*"
-        [[ -z "$msg" ]] && msg="failed"
-        echo "$sh_arg0: $msg" >&2
-        exit 1
-}
-
-[[ $# -eq 1 ]] || fail "missing arguments"
-
-# CTFCONVERT may contain a wildcard, so expand it out:
-ctfconvert=$(echo ${CTFCONVERT})
-[[ -x ${ctfconvert} ]] || fail "could not find ctfconvert at $CTFCONVERT"
-
-echo "Converting $1"
-$ctfconvert -L VERSION -o $1.ctf $1
-[[ $? -ne 0 ]] && exit 1
-mv $1.ctf $1
-exit 0
diff --git a/rules.mak b/rules.mak
index 81bfc03..e7b01c5 100644
--- a/rules.mak
+++ b/rules.mak
@@ -1,3 +1,6 @@
+#
+# Copyright (c) 2019, Joyent, Inc.
+#
 
 # Don't use implicit rules or variables
 # we have explicit rules for everything
@@ -25,9 +28,11 @@ QEMU_DGFLAGS += -MMD -MP -MT $@ -MF $(*D)/$(*F).d
 
 LINK = $(call quiet-command,$(CC) $(QEMU_CFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $(1) $(LIBS),"  LINK  $(TARGET_DIR)$@")
 
-CTFMERGE_CMD = $(call quiet-command, $(CTFMERGE) -L VERSION -o $@ $(1),"  CTFMERGE  $(TARGET_DIR)$@")
-
-CTFCONVERT_CMD = $(call quiet-command, CTFCONVERT=$(CTFCONVERT) find ../ -type f -name '*.o' -exec ../ctf.sh '{}' \;)
+#
+# Quite a few components are empty files or not compiled with CTF (such as
+# libpng).
+#
+CTFCONVERT_CMD = $(call quiet-command, $(CTFCONVERT) -m -L VERSION $@," CTFCONVERT  $(TARGET_DIR)$@")
 
 ifeq ($(TRACE_BACKEND),dtrace)
 ifneq ($(strip $(CONFIG_SOLARIS)),)
